tests/net/lib/mdns_responder/src/main.c

```diff
@@ -0,0 +1,445 @@
+
+	if (data->mac_addr[2] == 0x00) {
+		/* 00-00-5E-00-53-xx Documentation RFC 7042 */
+		data->mac_addr[0] = 0x00;
+		data->mac_addr[1] = 0x00;
+		data->mac_addr[2] = 0x5E;
+		data->mac_addr[3] = 0x00;
+		data->mac_addr[4] = 0x53;
+		data->mac_addr[5] = 0x01;
+	}
+
+	data->ll_addr.addr = data->mac_addr;
+	data->ll_addr.len = 6U;
+
+	return data->mac_addr;
+}
+
+static void net_iface_init(struct net_if *iface)
+{
+	uint8_t *mac = net_iface_get_mac(net_if_get_device(iface));
+
+	net_if_set_link_addr(iface, mac, sizeof(struct net_eth_addr),
+			     NET_LINK_ETHERNET);
+}
+
+static int sender_iface(const struct device *dev, struct net_pkt *pkt)
+{
+	struct net_ipv6_hdr *hdr;
+
+	if (!pkt->buffer) {
+		return -ENODATA;
+	}
+
+	if (test_started) {
+		hdr = NET_IPV6_HDR(pkt);
+
+		if (net_ipv6_addr_cmp_raw(hdr->dst, mdns_server_ipv6_addr)) {
+			if (responses_count < MAX_RESP_PKTS) {
+				net_pkt_ref(pkt);
+				response_pkts[responses_count++] = pkt;
+				k_sem_give(&wait_data);
+			}
+		}
+	}
+
+	return 0;
+}
+
+static struct dummy_api net_iface_api = {
+	.iface_api.init = net_iface_init,
+	.send = sender_iface,
+};
+
+#define _ETH_L2_LAYER DUMMY_L2
+#define _ETH_L2_CTX_TYPE NET_L2_GET_CTX_TYPE(DUMMY_L2)
+
+NET_DEVICE_INIT_INSTANCE(net_iface1_test,
+			 "iface1",
+			 iface1,
+			 NULL,
+			 NULL,
+			 &net_iface1_data,
+			 NULL,
+			 CONFIG_KERNEL_INIT_PRIORITY_DEFAULT,
+			 &net_iface_api,
+			 _ETH_L2_LAYER,
+			 _ETH_L2_CTX_TYPE,
+			 127);
+
+static void *test_setup(void)
+{
+	struct net_if_addr *ifaddr;
+	int idx;
+
+	memset(response_pkts, 0, sizeof(response_pkts));
+	memset(services, 0, sizeof(services));
+	memset(records, 0, sizeof(records));
+
+	responses_count = 0;
+
+	/* Cross assign records and buffers to entries for allocation */
+	for (int i = 0; i < EXT_RECORDS_NUM; ++i) {
+		services[i].record = &records[i];
+
+		records[i].instance = services[i].instance;
+		records[i].service = services[i].service;
+		records[i].proto = services[i].proto;
+		records[i].domain = services[i].domain;
+		records[i].text = services[i].text;
+		records[i].port = &services[i].port;
+	}
+
+	mdns_responder_set_ext_records(records, EXT_RECORDS_NUM);
+
+	/* The semaphore is there to wait the data to be received. */
+	k_sem_init(&wait_data, 0, UINT_MAX);
+
+	iface1 = net_if_get_by_index(1);
+
+	zassert_not_null(iface1, "Iface1 is NULL");
+
+	((struct net_if_test *) net_if_get_device(iface1)->data)->idx =
+		net_if_get_by_iface(iface1);
+
+	idx = net_if_get_by_iface(iface1);
+	zassert_equal(idx, 1, "Invalid index iface1");
+
+	zassert_not_null(iface1, "Interface 1");
+
+	ifaddr = net_if_ipv6_addr_add(iface1, &ll_addr,
+				      NET_ADDR_MANUAL, 0);
+
+	net_ipv6_nbr_add(iface1, &sender_ll_addr, net_if_get_link_addr(iface1), false,
+			 NET_IPV6_NBR_STATE_STATIC);
+
+	if (!ifaddr) {
+		zassert_not_null(ifaddr, "Failed to add LL-addr");
+	} else {
+		/* we need to set the addresses preferred */
+		ifaddr->addr_state = NET_ADDR_PREFERRED;
+	}
```

Comment:
We can simplify this to:
```
zassert_not_null(ifaddr, "Failed to add LL-addr");

/* we need to set the addresses preferred */
ifaddr->addr_state = NET_ADDR_PREFERRED;
```
because if the zassert fails, then the program stops so there is no need to have the if() else() branches.

Reply:
Done